code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -183,22 +199,132 @@
 		return src.close()
 	return src.open()
 
+/obj/structure/closet/proc/add_lock(var/obj/item/weapon/circuitboard/airlock/E, var/mob/user)
+	if(has_lock_type && !electronics && E && E.icon_state != "door_electronics_smoked")
+		playsound(src, 'sound/items/Screwdriver.ogg', 100, 1)
+		user.visible_message("[user] is installing electronics on \the [src].", "You start to install electronics into \the [src].")
+		if(do_after(user, src, 40))
+			var/obj/structure/closet/new_closet
+			new_closet = change_type(has_lock_type)
+
+			if(new_closet)
+				to_chat(user, "<span class='notice'>You installed the electronics!</span>")
+				user.drop_item(E, new_closet, force_drop = 1)
+				new_closet.electronics = E
+				E.installed = 1
+
+				if(E.one_access)
+					new_closet.req_access = null
+					new_closet.req_one_access = E.conf_access
+				else
+					new_closet.req_access = E.conf_access
+
+				new_closet.locked = 0
+				new_closet.update_icon()
+			else
+				//Should not happen
+				to_chat(user, "<span class='notice'>Wierd, the electronics won't fit.</span>")
+	else if(!has_lock_type)
+		to_chat(user, "<span class='notice'>There's no slot for the electronics</span>")
+
+
+/obj/structure/closet/proc/remove_lock(var/mob/user)
+	if(has_lockless_type)
+		playsound(src, 'sound/items/Screwdriver.ogg', 100, 1)
+		user.visible_message("[user] is removing \the [src]'s electronics.", "You start removing \the [src]'s electronics.")
+		if(do_after(user, src, 40))
+			var/obj/structure/closet/new_closet
+			new_closet = change_type(has_lockless_type)
+
+			if(new_closet)
+				new_closet.dump_electronics()
+				new_closet.broken = 0
+				new_closet.locked = 0
+				new_closet.update_icon()
+			else
+				//Should not happen
+				to_chat(user, "<span class='notice'>Weird, you can't get the electronics out.</span>")
+	else
+		to_chat(user, "<span class='notice'>You can't get the electronics out</span>")
+
+/obj/structure/closet/proc/dump_electronics()
+	var/obj/item/weapon/circuitboard/airlock/E
+	if (!electronics)
+		E = new/obj/item/weapon/circuitboard/airlock(loc)
+		if(req_access && req_access.len)
+			E.conf_access = req_access
+		else if(req_one_access && req_one_access.len)
+			E.conf_access = req_one_access
+			E.one_access = 1
+	else
+		E = electronics
+		electronics = null
+		E.forceMove(loc)
+		E.installed = 0
+
+	if (broken == 1)
+		E.icon_state = "door_electronics_smoked"
+
+	req_access = null
+	req_one_access = null
+
+	return E
+
+
+
+// Might come handy for painting crates and lockers some day.
+// Using it to change from secure to non secure lockers for now
+/obj/structure/closet/proc/change_type(var/new_type)
+	if(new_type)//Ensure it's not null
```

Comment:
You're already checking whether `has_lock_type` or `has_lockless_type` is null in the callers of this proc, so this is not necessary. Maybe even counterproductive as it will silently swallow the error if someone calls it with a null parameter one day, instead of runtiming. I'd say get rid of this or turn it into an `ASSERT()`